cmd/earthly/subcmd/satellite_cmds.go

```diff
@@ -987,8 +988,30 @@ func (a *Satellite) actionUpdate(cliCtx *cli.Context) error {
 		return errors.Errorf("not a valid platform: %q", platform)
 	}
 
+	if sat.State != cloud.SatelliteStatusSleep {
+		a.cli.Console().Printf("")
+		a.cli.Console().Printf("The satellite must be asleep to start the update.")
+		a.cli.Console().Printf("Putting the satellite to sleep will interrupt any running builds.")
+		a.cli.Console().Printf("")
+		answer, err := common.PromptInput(cliCtx.Context, "Would you like to put it to sleep now? [y/N]: ")
+		if err != nil {
+			return errors.Wrap(err, "failed to read permission")
+		}
+		answer = strings.ToLower(answer)
```

Comment:
nit: I think we should refactor this into a common `IsResponseYes(s string) bool` function, then make use of it here and https://github.com/earthly/earthly/blob/b337b88e4e78827c0566d009161ed9dee95b0a87/cmd/earthly/subcmd/project_cmds.go#L183